include/lug/System/Exception.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include <string>
+#include <lug/System/Export.hpp>
+
+namespace lug {
+namespace System {
+
+class LUG_SYSTEM_API Exception {
+public:
+    Exception(const char *typeName, const std::string &description, const char* file, const char* function, long line);
+    Exception(Exception&& other);
+    Exception(const Exception& other);
+    Exception& operator=(const Exception& other);
+    Exception& operator=(const Exception&& other);
+    ~Exception() throw() {}
+
+    const std::string&              getTypeName() const;
```

Comment:
You don't need to have many spaces between the type and the member function name. (Guidelines)